### ISSUE TITLE
[UX/UI] Button to get the current date in a date/time field is buggy and no date is displayed in the date/time main field box after manually selecting a date in QField 3.0.7 (35e3ee)

### ISSUE THREAD

**Issue by nicolas-sitylb** [opengisch/QField#4803]
### Describe the issue
The small calendar icon on the right of a field box allowing the user to insert the current date in a date/time field is buggy.
It says it has written the current date but the date displayed in the field is `1900-01-01 12:00:00+CET` and when committing the changes, the database issues an error.

E.g. during an update:
```
ERROR:  invalid input syntax for type date: "QCET ISO DpmCETe" at character xx
STATEMENT:  UPDATE "public"."some_table" SET "date_field"='QCET ISO DpmCETe',"created_at"='2023-11-24T11:14:37.371',"updated_at"='2023-11-24T11:14:37.369' WHERE "id"=10
```

Whereas, the complete calendar opening when hitting the main field box itself (i.e. not the small calendar icon on the right of the field box) is working perfectly. The only thing with the main calendar, it that the selected date is never displayed in the field box once the OK button has been clicked. It "visually" stays empty, but when committing the value, everything is correctly inserted/updated. It's only a display issue I guess.

### Reproduction steps
1. Add or Edit a feature.
2. Click on the small calendar icon on the right of the main field box containing a date/time field to get the current date 
3. Save the feature / update.

### Expected behavior
The expected behavior is to display the _correct_ current date (now) in the main field box when hitting the small calendar icon on its right.

### Observed behavior
See "Describe the issue" above.

### Mobile (please complete the following information)
 - Device: HAPPY-TAB-8_V1.1
 - OS: Android 10
 - QField version: 3.0.7 (35e3ee)

### Additional information
* Problem started happening recently, didn't happen in an older version of QField: N/A
* Problem can be reliably reproduced, doesn't happen randomly: YES
* Problem happens with all files and projects, not only some files or projects: YES

**Comment by nirvn** [comment]
@nicolas-sitylb , would you have a test project to share?

**Comment by nicolas-sitylb** [comment]
@nirvn I noticed that the problem was mostly due to a mismatching in the way the date was formatted between:
1) a PG database     
2) a QGIS server project connected to the database     
3) and the QGIS client project (used by QField) - consuming the QGIS server project through the WFS protocol

On the server side, there is a PG database having a `date` field. 

In a QGIS server project connected to that database, there is a `date` field that was mapped manually (in the Widget Type > Field format within the "Attribute Form" in the "Layer Properties") using:
- in a first attempt : "dd.MM.YYYY"     
- and in a second attempt : "ISO Date Time"     

In both those cases, the same `date` field in the QGIS client was set to "YYYY-MM-dd". In both those cases QField was inserting a totally wrong date when hitting the small calendar button.

I have the feeling that QGIS, in a way,  _SHOULD NOT_ allow the user to "override" the date format of the database because when I set the date format (both in the QGIS server and client projects) to a custom value corresponding _exactly_ to the format it has in the PG database; "YYYY-MM-dd" then everything but the update (addressed by https://github.com/opengisch/QField/pull/4804) goes fine.

I may be wrong in my interpretation, but it's only when there is a mismatch between the database, the QGIS server project and the QGIS client project in the way the date is formatted that things behave in a weird (unpredictable?) manner in QField.    
Also, I don't know of the use of the WFS protocol may interfere in that pipeline.

If I find the time, I'll try to reproduce the error using a small Docker stack.

BTW: if the date format is everywhere something not including a **time** definition (hours-minutes-seconds), the main calendar should not propose to select it but it does.
